scripts/orig_scripts/txt2img.py

```diff
@@ -240,11 +248,15 @@ def forward(self, x, sigma, uncond, cond, cond_scale):
 
     start_code = None
     if opt.fixed_code:
-        start_code = torch.randn([opt.n_samples, opt.C, opt.H // opt.f, opt.W // opt.f], device=device)
+        start_code = torch.randn(
```

Comment:
for what it's worth, this will slightly slow down startup for CUDA users.  
it's also not necessary to wrap device in `torch.device()`.

how about only falling back to CPU if you're on MPS:

```python
rand_size = [opt.n_samples, opt.C, opt.H // opt.f, opt.W // opt.f]
# https://github.com/CompVis/stable-diffusion/issues/25#issuecomment-1229706811
# MPS random is not currently deterministic w.r.t seed, so compute randn() on-CPU
torch.randn(rand_size, device='cpu').to(device) if device.type == 'mps' else torch.randn(rand_size, device=device)
```

Reply:
you can simplify `rand_size` initialization if you move the `shape` initialization from line 271 earlier:

```python
shape = [opt.C, opt.H // opt.f, opt.W // opt.f]
```

```python
rand_size = [opt.n_samples, *shape]
# https://github.com/CompVis/stable-diffusion/issues/25#issuecomment-1229706811
# MPS random is not currently deterministic w.r.t seed, so compute randn() on-CPU
start_code = torch.randn(rand_size, device='cpu').to(device) if device.type == 'mps' else torch.randn(rand_size, device=device)
```

Reply:
`torch.randn(rand_size, device=device)` is a bug I introduced (sorry, I'm new to this). On the CompVis issue for Mac support [someone pointed this out](https://github.com/CompVis/stable-diffusion/issues/25#issuecomment-1229706811).

Reply:
Arg... I am wrong. I didn't introduce this code. It is device=device in the CompVis upstream. I was sure that I was the one who changed it to that and that it was originally device="cpu"... That means if we make this change it will affect everyone and it will break all seeds. Maybe the source of the random seed should be it's own flag.

Reply:
@Birch-san, I totally see what you mean. Yeah, I'll make this change.

Reply:
I think we need a variable like `rng_device` which says where our RNG source lives. for CPU and CUDA users it'll be same-device. for MPS users it'll be CPU.